preppal-be/routes/authApi.ts

```diff
@@ -0,0 +1,56 @@
+const expressAuthApi = require("express");
+const routerAuthApi = expressAuthApi.Router();
+const jwtAuthApi = require("jsonwebtoken");
+const configAuthApi = require("../configs/secrets.ts")
+
+const UserAuth = require('../models/user.ts');
+
+/**
+ * GET - Get all accounts
+ */
+routerAuthApi.get("/", async (req, res) => {
+    try {
+        const users = await UserAuth.find().select("-password");
+        res.json(users);
+    } catch (error) {
+        console.error(error.message);
+        res.status(500).send("Server error");
+    }
+});
+
+/**
+ * POST - Login to an account
+ */
+routerAuthApi.post("/", async(req, res) => {
+    try {
+        const { username, password } = req.body;
+        let user = await UserAuth.findOne({ username: username });
+        if (!user) {
+            return res.status(400).json({ errors: [{ msg: "Invalid username or password!" }] });
+        }
+
+        if (password != user.password) {
```

Comment:
this can be placed combined into one if statement with the one above it...

if (!user || !password != user?.password)

Reply:
Fixed. doing `if (!user || password != user.password)` will work.
No need to put a ? since it will fail before that.
Tested in postman with only username and only password.